zerver/data_import/slack.py

```diff
@@ -357,25 +358,22 @@ def users_to_zerver_userprofile(
         else:
             bot_type = None
 
-        userprofile = UserProfile(
+        userprofile_dict = build_user_profile(
+            avatar_source=avatar_source,
+            date_joined=timestamp,
+            delivery_email=email,
+            email=email,
             full_name=get_user_full_name(user),
+            id=user_id,
             is_active=not user.get("deleted", False) and not user["is_mirror_dummy"],
+            role=role,
             is_mirror_dummy=user["is_mirror_dummy"],
-            id=user_id,
-            email=email,
-            delivery_email=email,
-            avatar_source=avatar_source,
+            realm_id=realm_id,
+            short_name=user["name"],
+            timezone=timezone,
             is_bot=is_bot,
-            role=role,
             bot_type=bot_type,
-            date_joined=timestamp,
-            timezone=timezone,
-            last_login=timestamp,
```

Comment:
It looks like we lost `last_login` here -- is that intended?

I'm not sure making that match `date_joined` is at all correct, but we should explicitly look at whether there's a better value to use. Possibly "None" would be the right value for all import tools, which I think is what this is switching to?